lms/djangoapps/commerce/api/v1/views.py

```diff
@@ -64,10 +73,17 @@ class OrderView(APIView):
     """ Retrieve order details. """
 
     authentication_classes = (SessionAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticatedOrActivationOverridden,)
```

Comment:
Yes, you were totally right. `ModelBackend`, the authentication backend we use checks is_active flag on the authenticated user and will return `AnonymousUser` if it is `False`.

Using the Otto receipt page things seem to work just fine without these changes to `OrderView`. Obviously this makes sense because this view does not get called from the Otto receipt page.

My vote would be to exclude these changes since the LMS receipt page is deprecated and really should actually be (have been) removed.